### PR TITLE
Refactor: Code Cleanup ♻️ 

### PR DESCRIPTION
**Pull request by jyash97**
**Changes**
- Separate props logic from `Editor` & `RSWrapper` to a single file.
- Use consumer in RSWrapper instead of passing props from Editor.
- Separate Logic for ReactiveList `onData` - `RenderResults`
   - We can add different type to change the layout of ReactiveList now.
- Separate `JSON Editor` & `Delete JSON` to separate Component

### CODE REVIEW

File: components/SearchSandbox/components/RSWrapper.js [hunk 1]
```diff
@@ -439,11 +445,9 @@ export default class RSWrapper extends Component {
 	};
 
 	renderFormItem = (item, name) => {
-	const { componentProps, mappings } = this.props;
+		const { componentProps } = this.props;
 		let FormInput = null;
-		const value = componentProps[name] !== undefined
-				? componentProps[name]
-				: item.default;
+		const value = componentProps[name] !== undefined ? componentProps[name] : item.default;
```
Comment (metagrover) [inline components/SearchSandbox/components/RSWrapper.js]: It should be:

```
componentProps[name] === undefined ? item.default : componentProps[name];
```

File: components/SearchSandbox/utils/getComponentProps.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import RenderResults from '../components/RenderResults';
+import { generateFieldWeights, generateDataField } from './dataField';
+
+function dataSearchProps({ componentProps, mappings }) {
```
Comment (metagrover) [inline components/SearchSandbox/utils/getComponentProps.js]: We should rename these functions to:

```
get<Component-name>Props
```

File: components/SearchSandbox/components/RSWrapper.js (hunk 1 above)
Comment (jyash97) [reply components/SearchSandbox/components/RSWrapper.js]: Yeah will fix ths 👍

File: components/SearchSandbox/utils/getComponentProps.js (hunk 2 above)
Comment (jyash97) [reply components/SearchSandbox/utils/getComponentProps.js]: Sure

**Comment by jyash97** [comment]
Besides this everything else looks good to go ?

**Comment by metagrover** [comment]
Yes, looks good. This definitely simplifies the generateDataField logic. 👍

**Comment by jyash97** [comment]
Cool 👍 Will do the changes.
After this we can check on #84 & #85

**Pull request merged**